app/controllers/incoming_messages_controller.rb

```diff
@@ -10,36 +10,33 @@ def create
     end
 
     account = Account.find_by(slug: params.fetch(:account))
-    email = Mail::Address.new params.fetch(:email)
-    author = MessageAuthor.new(account, email)
-    attachment = params[:attachment]
+    person = Person.find_or_create_by_addr(
+      Mail::Address.new(params.fetch(:email))
+    )
 
-    begin
-      ActiveRecord::Base.transaction do
-        conversation = Concierge.new(account, params).find_conversation
+    person.save!
 
-        @message = author.compose_message(conversation, params.fetch(:content))
-        @message.save!
+    @message = MessageFactory.build(
+      account: account,
+      person:  person,
+      content: params.fetch(:content)
+    )
 
-        @message.attachments.create(file: attachment) if attachment.present?
-
-        if Rails.env.production?
-          Customerio.client.track(account.owner.id, 'received_message')
-        end
+    if @message.save
+      if Rails.env.production?
+        Customerio.client.track(account.owner.id, 'received_message')
+      end
 
-        respond_to do |format|
-          format.html
-          format.json do
-            render json: @message,
-              status: :created,
-              callback: params[:callback],
-              content_type: content_type
-          end
+      respond_to do |format|
+        format.json do
+          render json: @message,
+            status: :created,
+            callback: params[:callback],
+            content_type: content_type
         end
       end
-    rescue ActiveRecord::RecordInvalid
+    else
       respond_to do |format|
-        format.html
```

Comment:
I'm not sure about this but i think we need this in the case of folks wiring up an HTML form directly to Helpful. Let's keep the html response here for now.